### PR TITLE
Process sudo

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

File: tool/state-processor/src/sudo/mod.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// darwinia
+use crate::*;
+
+impl<S> Processor<S> {
+	pub fn process_sudo(&mut self) -> &mut Self {
+		// Storage items.
+		// https://github.dev/darwinia-network/substrate/blob/darwinia-v0.12.5/frame/sudo/src/lib.rs#L268
+		//
+		// The new sudo key will be set on the genesis side.
```
[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply tool/state-processor/src/sudo/mod.rs:9]: https://github.com/darwinia-network/darwinia-2.0/blob/459becae40f6c83bec618ba614d23e3df0a345bc/node/src/chain_spec/pangolin.rs#L277

[the remark this answers is not in the document]
Comment (boundless-forest) [reply tool/state-processor/src/sudo/mod.rs:9]:
```rs
mem::swap(&mut self.shell_state.map, &mut self.shell_chain_spec.genesis.raw.top);
```

The whole state map comes from the shell state(processed.json), I think the sudo key ALITH does not inject into the new state.

[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply tool/state-processor/src/sudo/mod.rs:9]: The shell state comes from:
https://github.com/darwinia-network/darwinia-2.0/blob/459becae40f6c83bec618ba614d23e3df0a345bc/tool/state-processor/src/processor.rs#L35

[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply tool/state-processor/src/sudo/mod.rs:9]: Also, they are already swapped:
https://github.com/darwinia-network/darwinia-2.0/blob/459becae40f6c83bec618ba614d23e3df0a345bc/tool/state-processor/src/processor.rs#L43-L46

File: node/src/chain_spec/darwinia.rs [hunk 2]
```diff
@@ -188,7 +188,7 @@ pub fn genesis_config() -> ChainSpec {
 				treasury: Default::default(),
 
 				// Utility stuff.
-				sudo: Default::default(),
+				sudo: SudoConfig { key: Some(array_bytes::hex_n_into_unchecked(ALITH)) },
```
Comment (boundless-forest) [inline node/src/chain_spec/darwinia.rs:191]: This reminds me of the `AssetsConfig` set in the state processor, we might not need them there.

**Pull request merged**